detection-rules/attachment_office365_image.yml

```diff
@@ -7,18 +7,18 @@ authors:
   - twitter: "vector_sec"
 source: |
   type.inbound
-  and length(body.html.inner_text) < 100
-  and any(attachments, .file_extension in~ ('png','jpg','jpeg')
+  and length(body.html.inner_text) < 300
+  and any(attachments, .file_extension in~ ('bmp','png','jpg','jpeg')
       and any(file.explode(.),
           (
               (
-                  any(.scan.strings.strings, strings.ilike(., "*microsoft*", "*office365*", "*office*")) and
-                  any(.scan.strings.strings, strings.ilike(., "*password*", "*expire*", "*expiry*"))
+                  any(.scan.strings.strings, strings.ilike(., "*password*", "*microsoft*")) and
+                  any(.scan.strings.strings, strings.ilike(.,  "*expire*", "*expiry*","*office365*", "*office*"))
               )
           or
               (
-                  strings.ilike(.scan.ocr.raw, "*microsoft*", "*office365*", "*office*") and
-                  strings.ilike(.scan.ocr.raw, "*password*", "*expire*", "*expiry*")
+                  strings.ilike(.scan.ocr.raw, "*microsoft*", "*password*") and
+                  strings.ilike(.scan.ocr.raw,  "*office365*", "*office*", "*expire*", "*expiry*")
```

Comment:
this right? `(microsoft or password) and (office or expire or expiry)`
also `office365` here is redundant with `office`

Reply:
Yes that's correct, the way they were split up before wasn't ensuring that "microsoft" or "password" was present and was firing on images containing "office" and "expire".  Requiring either 'Microsoft' or 'password' in addition to one of the others removed the FP's and more directly captured the intent imo.

Nice catch on the redundancy, since it was a community authored rule and we adopted that redundancy I just left it in. But I'll update it tomorrow!! Thanks!!! 👍

Reply:
so if you have a JPG with the string `microsoft office` in the strings, it'll flag? That sounds like an FP storm

Reply:
I thought it would be as well, it performs well in our rules repo, and other instances without FP's. That said our test environment can be biased.  I'm happy to redo the rule altogether,  folks were asking to loosen it as it was missing detectable opportunities like:

![image](https://user-images.githubusercontent.com/6809735/234732846-3df450e9-e670-49ee-9ecb-bc9e97918f98.png)

I originally had it split like this, but it was still missing the FN's. I'll revisit tomorrow! Appreciate the eyes!
`
        (
            (   any(.scan.strings.strings, strings.ilike(., "*password*", "*microsoft*")) and
                any(.scan.strings.strings, strings.ilike(.,  "*office365*", "*office*")) and
                any(.scan.strings.strings, strings.ilike(.,  "*expire*", "*expiry*"))
            )
        or
            (
                strings.ilike(.scan.ocr.raw, "*microsoft*", "*password*") and
                strings.ilike(.scan.ocr.raw,  "*office365*", "*office*") and
                strings.ilike(.scan.ocr.raw, "*expire*", "*expiry*")
            )
        )
      '